src/components/EventCardsGridContainer/index.js

```diff
@@ -77,23 +88,99 @@ class EventCardsGrid extends React.Component {
     } = nextProps;
 
     if (eventStatusIndex !== this.props.eventStatusIndex
-      || sortBy !== this.props.sortBy
-      || syncBlockNum !== this.props.syncBlockNum) {
-      this.executeGraphRequest(eventStatusIndex, sortBy);
+      || sortBy !== this.props.sortBy) {
+      this.executeGraphRequest(eventStatusIndex, sortBy, LIMIT, SKIP);
+      this.setState({ skip: 0 });
+    }
+    if (syncBlockNum !== this.props.syncBlockNum && this.props.syncBlockNum !== 0) {
+      this.setState({ hint: true });
+      setTimeout(() => {
+        this.setState({ hint: false });
+      }, 3000);
     }
   }
 
-  render() {
+  loadMoreOracles = () => {
```

Comment:
this seems to be an exact duplicate of `executeGraphRequest()`. except for incrementing the skip. why not reuse that function instead of having duplicate code?